clock/scheduledJobs/jobCleanup.js

```diff
@@ -16,40 +16,110 @@ const Promise = require('bluebird');
 kue.Job.prototype.removeAsync = Promise.promisify(kue.Job.prototype.remove);
 
 /**
  * Execute the call to clean up jobs
  *
+ * @param {Number} window - the number of jobs to delete in each batch
+ * @param {Number} delay - the delay, in ms, before completed jobs should be deleted
  * @returns {Promise}
  */
-function execute() {
+function execute(window, delay) {
+  const now = Date.now();
+  let removedJobCount = 0;
+  let skippedJobCount = 0;
+
   if (featureToggles.isFeatureEnabled('instrumentKue')) {
     console.log('[KJI] Ready to remove completed jobs');
   }
 
-  return rangeByStateAsync('complete', 0, -1, 'asc') // get all completed jobs
-  .then((jobs) =>
-    Promise.all(jobs.map((job) =>
-      job.removeAsync()
-      .then(() => {
-        if (featureToggles.isFeatureEnabled('instrumentKue')) {
-          console.log(`[KJI] Removed completed job ${job.id}`);
-        }
-      })
-      .catch((err) => {
-        console.log(`Error removing job ${job.id}`, err);
-        return Promise.reject(err);
-      })
-    ))
-  )
-  .catch((err) => {
-    console.log('Error getting completed jobs from queue', err);
-    return Promise.reject(err);
+  return deleteNextNJobs(window)
+  .then(() => {
+    if (featureToggles.isFeatureEnabled('instrumentKue')) {
+      console.log(`[KJI] Removed ${removedJobCount} jobs`);
+    }
   });
+
+  function deleteNextNJobs(n) {
+    if (n === 0) return Promise.resolve();
+    const from = skippedJobCount;
+    const to = skippedJobCount + n - 1;
+
+    // get n jobs
+    return rangeByStateAsync('complete', from, to, 'asc')
+    .catch((err) => {
+      console.log('Error getting completed jobs from queue', err);
+      return Promise.reject(err);
+    })
+
+    // delete n jobs
+    .then((jobs) =>
+      Promise.all(jobs.map((job) => {
+        if (now - job.updated_at < delay) {
+          return 'skipped';
+        } else {
+          return job.removeAsync()
+          .catch((err) => {
+            console.log(`Error removing job ${job.id}`, err);
+            return Promise.reject(err);
+          });
+        }
+      }))
+    )
+
+    // count skipped jobs
+    .then((results) => {
+      const skippedCount = results.reduce((count, result) =>
+        result === 'skipped' ? count + 1 : count,
+      0);
+      skippedJobCount += skippedCount;
+      removedJobCount += results.length - skippedCount;
+      return Promise.resolve(results);
+    })
+
+    // get and delete next n jobs
+    .then((results) => {
+      if (results.length < n) {
+        return Promise.resolve();
+      } else {
+        return deleteNextNJobs(n);
+      }
+    });
+  }
+
 } // execute
 
+/**
+ * Send the job to the worker or execute directly
+ */
+function enqueue() {
+  if (featureToggles.isFeatureEnabled('enableWorkerProcess')) {
+    const jobWrapper = require('../../jobQueue/jobWrapper');
+    const jobType = require('../../jobQueue/setup').jobType;
```

Comment:
require('../../jobQueue/setup') has already been required at the top. You could just use that.